src/test/java/io/anserini/server/SearchServiceTest.java

```diff
@@ -116,4 +118,50 @@ public void testInvalidIndex() {
     assertThrows("InvalidIndex: Constructing SearchService with index 'nonexistent-index' should throw RuntimeException",
         RuntimeException.class, () -> new SearchService("nonexistent-index"));
   }
+
+  @Test
+  public void testShardedSearch() throws Exception {
+    String identifier = "msmarco-v2.1-doc-segmented.arctic-embed-l.hnsw-int8";
+    ShardInfo shardGroup = ShardInfo.fromIdentifier(identifier);
+
+    List<Map<String, Object>> results = SearchService.searchSharded(
+        shardGroup, "test query", 10, null, null, null,
+        index -> new SearchService(index));
+
+    assertNotNull(String.format("Expected non-null results for sharded search with identifier '%s'", identifier),
```

Comment:
This and below... I'm fine with longish lines.
People have large monitors these days...